src/libpmemobj/memops.h

```diff
@@ -39,7 +40,53 @@ struct user_buffer_def {
 	size_t size;
 };
 
-struct operation_context;
+enum operation_state {
```

Comment:
It would be better if we kept the structs private and instead relied on some inter-module API if necessary.

Reply:
ok....